src/App.js

```diff
@@ -1,26 +1,75 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+import React, { Component } from 'react'
+import Header from './Header.js'
+import ImageList from './ImageList.js'
+import hornData from './data.js'
+
+
+
+export default class App extends Component {
+  state = {
+    title: '',
+    description: '',
+    horns: 1,
+    keyword: ''
+  };
+
+  handleChange = (event) => {
+    this.setState({
+      horns: event.target.value
+    });
+  }
+
+  handleInputChange = (event) => {
+    this.setState({
+      keyword: event.target.value
+    })
+  }
+
+  handleNameChange = (event) => {
+    this.setState({
+      title: event.target.value
+    })
+  }
+
+  handleDescriptionChange = (event) => {
+    this.setState({
+      description: event.target.value
+    })
+  }
 
-export default App;
+
+
+  render() {
+    const filteredCreatures = hornData
+    .filter(item => item.horns >= this.state.horns)
+    .filter(item => item.keyword.includes(this.state.keyword))
+    .filter(item => item.title.includes(this.state.title))
+
```

Comment:
Your description filter isn't working because you haven't include it here.
    .filter(item => item.description.includes(this.state.description))